acceptancetests/deploy_stack.py

```diff
@@ -1154,3 +1236,15 @@ def wait_for_state_server_to_shutdown(host, client, instance_id, timeout=60):
         else:
             raise Exception(
                 '{} was not deleted:'.format(instance_id))
+
+
+def get_client_and_test(test, args):
```

Comment:
Not a fan of this name, you're not getting a client and getting a test. Perhaps something like test_on_controller(), run_on_controller(), run_on_bootstrapped_controller() or something along those lines.

I'm on the fence wherever passing in a test callable is a nice way to do it. It reduces flexibility, but do we really care?
At any rate if this function continues to exist it really needs docstrings explaining what it does and what 'test' is (and it's requirements/constraints are.